### ISSUE TITLE
width of the file tabs in the notepad++ window

### ISSUE THREAD

**Issue by juanfmirandam** [notepad-plus-plus/notepad-plus-plus#3332]
### Description of the Issue
Hello! this is my first post here.  When one has 20 files open in different tabs in Notepad++, the program displays the full file name in the tab for each file, which means that the tab takes a width dependent on its file name. Google Chrome for example does not do this. It autoadjusts the size of each tab depending on the number of open tabs. I wonder if Notepad++ could in the future include such a feature .

Another related feature would be to make it possible to search among the list of open tabs for a word in the filename (that is, the tab name) or even the content. 

Having said that, I love Notepad++. Incredibly light, flexible and powerfull.

**Comment by Kavindi183** [comment]
Hello!This is my first post here too.i checked about this issue.(difference of file tabs width) and i would like to work on this issue.Is it OK to add this feature to notepad++ ?

**Comment by AngryGamer** [comment]
> the program displays the full file name in the tab for each file, which means that the tab takes a width dependent on its file name. 

This is not entirely accurate. Notepad++ doesn't use TCS_RAGGEDRIGHT so if a lone tab is pushed onto the next row, then its width will be stretched to take up the whole row.

**Comment by Kavindi183** [comment]
Please give me some guidance to start work on this issue.

**Comment by juanfmirandam** [comment]
It would be great if you do! though I have no clue how that would be done. I imagine something smart with tab size like what Google Chrome has.

**Comment by freezer2022** [comment]
Already possible in internet browsers : https://www.makeuseof.com/set-minimum-tab-width-firefox-chrome/

**Comment by dbareis** [comment]
Please let me know but if I'm wrong but I can't see any tab options to minimise the tab width and to set a maximum size that it can be.  What it does now is just ridiculous.  I'm liking npp more and more but currently the biggest annoyance of all is it's tab handling.

I can "live" with it until it goes over multiple tabs (even when it didn't have to) as per the picture below.  I'd rather tab text gets truncated than go to multiple lines, so I'd like an option to set the tab width to the text width but have a maximum size after which the text gets truncated.  Or as mentioned above similar to what a browser does anything would be better than what is happening now.

![Image](https://github.com/user-attachments/assets/e712efa1-84bb-44dc-b321-2f1648dabca2)

**Comment by alankilborn** [comment]
>What it does now is just ridiculous. 

IYO.  IMO it works just fine.

>even when it didn't have to) as per the picture below.

I think it _did_ have to (go to multiple rows), because 3 of those long-named files wouldn't fit in one row.

>truncation

This doesn't make users happy that have a lot of commonly-prefixed filenames.

**Comment by dbareis** [comment]
Apps have options to make people happy, otherwise why not use notepad?  I
was hoping you'd be maximising happiness rather than restricting it to a
pool of users.

---
Thanks
Dennis

**Comment by freezer2022** [comment]
> alanikborn: This doesn't make users happy that have a lot of commonly-prefixed filenames.

Yes, but ping-pong: on the other hand the current state doesn't make users happy that have not many of commonly-prefixed filenames. 

If there exist more than one pool of users / usage scenarios, for a given functionality, there definately should be a setting/switch to satisfy the both pools of users. Means that in the current case, there should be a min and max tab size setting available. And by the way this is a second thread wherein I noticed that you seem to be either unaware of / or intolerant towards other possible usage scenarios (the previous case was here: `https://github.com/notepad-plus-plus/notepad-plus-plus/issues/15615`), not sure about existence of other cases, as I don't follow all your posts in all threads.

**Comment by alankilborn** [comment]
> alankilborn seem to be either unaware of / or intolerant towards other possible usage scenarios

Neither.  I'm just expressing _my_ opinion, which is one of the functions for the "comment" trail for issues.

**Comment by freezer2022** [comment]
Fair and comprehensive opinions should be balanced and consider / mention also other usage scenarios involving other pools of users. Your opinion declined "truncation", as unwanted for some usage scenarios, and this is allright, what was unfair was that you didn't even agree on a part that "truncation" is helpful in some other usage scenarios. You have selfish / egoistic attitude in your opinions. Notepad++ is made for everyone. Obviously different folks want different things. It's obvious that tabs width should be a customizable option. I agree we don't want to overload a Notepad++ with a billion configurable settings but it seems this one is one worth having.

**Comment by dbareis** [comment]
Hi,

Truncation is one option and if implemented then front or tail truncation
should also be.  Font size reduction is another possibility.  But min max
is a good start that users can turn off or on.

---
Thanks
Dennis

**Comment by RISC-VVV** [comment]
> Hi,
> 
> Truncation is one option and if implemented then front or tail truncation
> should also be.  Font size reduction is another possibility.  But min max
> is a good start that users can turn off or on.
> […](#)

Adding a customizable option for label width seems worth having. When we set a fixed width label bar, our focus is on ensuring consistent label bar width to improve editing efficiency. At this point, whether the file name is displayed completely is not as important.

**Comment by dbareis** [comment]
The fact that tabs are padded out to sometimes the whole width of he bar is another annoyance I would like to be optional as it is extremely annoying to me at least.

**Comment by ArgonianVoter** [comment]
great now is there someway we can SHUT THIS OFF so i can FIND my tabs?! I didn't want this. if my filename is 999,999 characters who are you to tell me no? : ^)

**Comment by freezer2022** [comment]
> ArgonianVoter: 

What the h... are you talking about? The feature is disabled by default. And in case you enabled it by yourself on purpose or accidentaly, and are too inept to restore the settings, you should follow the simple instructions that are right above your comment or in the application itself.
